### DISCUSSION TITLE
Answered: Simplest unpaginated page equivalent?

### DISCUSSION THREAD

**Discussion by gitpushdashf** [uriyyo/fastapi-pagination#666] [Q&A] [answered]
Hi everyone,

I have the rather odd problem of needing to get rid of pagination, perhaps temporarily, without breaking existing frontend. So we want to have `items: list[Foo]` instead of say `list[Foo]`, for what was `Page[Foo]`.

I can go through and write `ListEndpointXResponse` classes for each one, or I can make a generic to replace `Page` with.

```python
from typing import Generic, TypeVar, Sequence
from pydantic.generics import GenericModel

T = TypeVar("T")


class UnpaginatedPage(GenericModel, Generic[T]):
    items: Sequence[T]
```

And this works great, except one issue. Let's say we're using SQLModel, and we have a different response model than the database object, itself. We want to convert to the `Read` version of that response.

This doesn't do that conversion. I've tried setting `Config`'s `orm_mode = True`, but no dice. I've looked through the fastapi-pagination code and fastapi's code to figure out what makes the `response_model` setting tick, but I can't figure it out.

Curious if anyone knows the answer to this. I feel like it's a pretty short bit of code to make it work.

For now, I'm manually converting my database objects to their read variants.

Thanks!

**Comment by uriyyo** [comment, marked as answer]
Hi @gitpushdashf,

I hope the example below will help you:
```py
from typing import Any, Generic, List, Optional, TypeVar

from fastapi import FastAPI
from pydantic import BaseModel
from pydantic.generics import GenericModel
from sqlalchemy import select
from sqlmodel import Field, Session, SQLModel, create_engine

engine = create_engine("sqlite:///database.db")


class User(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str
    email: str


SQLModel.metadata.drop_all(engine)
SQLModel.metadata.create_all(engine)

app = FastAPI()


@app.on_event("startup")
async def on_startup() -> None:
    with Session(engine) as session:
        session.add(
            User(
                id=1,
                name="John Doe",
                email="user@example.com",
            ),
        )
        session.commit()


class UserOut(BaseModel):
    id: int
    name: str
    email: str

    class Config:
        orm_mode = True


T = TypeVar("T")


class GenericListSchema(GenericModel, Generic[T]):
    items: List[T]


@app.get("/users", response_model=GenericListSchema[UserOut])
async def get_users() -> Any:
    with Session(engine) as session:
        users = session.scalars(select(User)).all()

        return {"items": users}
```

**Reply by gitpushdashf** [reply]
That's great! Thank you so much. I'll give that a try.

**Reply by gitpushdashf** [reply]
This works great. Thank you!

The trick is definitely to `return {"items": items}` instead of `return UnpaginatedPage(items=items)`.

**Reply by uriyyo** [reply]
Great, happy to hear that ✨

**Reply by uriyyo** [reply]
@gitpushdashf Could you please mark this discussion as answered?

**Reply by gitpushdashf** [reply]
I prefixed the title with `Answered:` -- is that a good way to do it? I didn't see any other means.

**Reply by uriyyo** [reply]
@gitpushdashf I converted it to `Q&A` discussion type and marked it as answered, thanks.
